plugins/spi/SPIOutput.cpp

```diff
@@ -486,6 +502,135 @@ uint8_t SPIOutput::P9813CreateFlag(uint8_t red, uint8_t green, uint8_t blue) {
   return ~flag;
 }
 
+
+void SPIOutput::IndividualAPA102Control(const DmxBuffer &buffer) {
+  // some detailed information on the protocol:
+  // https://cpldcpu.wordpress.com/2014/11/30/understanding-the-apa102-superled/
+  // Data-Struct
+  // StartFrame: 4Byte = 32Bits zeros (APA102_START_FRAME_BYTES)
+  // LEDFrame: 1Byte FF ; 3Byte color info (Blue, Green, Red)
+  // EndFrame: (n/2)bits; n = pixel_count
+
+  // latch_bytes = EndFrame ??
+  const uint8_t latch_bytes = CalculateAPA102LatchBytes(m_pixel_count);
+  // OLA_INFO << "latch_bytes " << static_cast<int>(latch_bytes);
+
+  // calculate dmx-start-address
+  const unsigned int first_slot = m_start_address - 1;  // 0 offset
+  // OLA_INFO << "first_slot " << static_cast<int>(first_slot);
+
+  // only do something if minimum 1pixel can be updated..
+  if (buffer.Size() - first_slot < APA102_SLOTS_PER_PIXEL) {
+    // not even 3 bytes of data, don't bother updating
+    return;
+  }
+
+  // We always check out the entire string length, even if we only have data
+  // for part of it
+  const unsigned int output_length = APA102_START_FRAME_BYTES +
+                                (m_pixel_count * APA102_SPI_BYTES_PER_PIXEL);
+  // OLA_INFO << "output_length " << static_cast<int>(output_length);
+  uint8_t *output = m_backend->Checkout(m_output_number, output_length,
+                                        latch_bytes);
+  // only update spi data if possible
+  if (!output)
```

Comment:
Can you add { } brackets around the return, so we don't have a dangling statement.